schema-registry/src/main/java/io/streamnative/pulsar/handlers/kop/schemaregistry/model/CompatibilityChecker.java

```diff
@@ -0,0 +1,185 @@
+        }
+        log.info("Compare schema against {} ids", idsToCheck);
+
+        CompletableFuture<List<Schema>>
+                res = schemaStorage.downloadSchemas(idsToCheck);
+
+        return res.thenApply((downloadedSchemas) -> {
+            return verify(schema.getSchemaDefinition(), schema.getType(), mode, downloadedSchemas);
+        });
+    }
+
+    public static boolean verify(String schemaDefinition, String type, Mode mode, List<Schema> allSchemas) {
+        if (allSchemas.isEmpty()) {
+            return true;
+        }
+        io.apicurio.registry.rules.compatibility.CompatibilityChecker checker = createChecker(type);
+        boolean onlyLatest = false;
+        CompatibilityLevel level;
+        switch (mode) {
+            case BACKWARD:
+                level = CompatibilityLevel.BACKWARD;
+                onlyLatest = true;
+                break;
+            case BACKWARD_TRANSITIVE:
+                level = CompatibilityLevel.BACKWARD_TRANSITIVE;
+                break;
+            case FORWARD:
+                level = CompatibilityLevel.FORWARD;
+                onlyLatest = true;
+                break;
+            case FORWARD_TRANSITIVE:
+                level = CompatibilityLevel.FORWARD_TRANSITIVE;
+                break;
+            case FULL:
+                level = CompatibilityLevel.FULL;
+                onlyLatest = true;
+                break;
+            case FULL_TRANSITIVE:
+                level = CompatibilityLevel.FULL_TRANSITIVE;
+                break;
+            default:
+                level = CompatibilityLevel.NONE;
+                onlyLatest = true;
+                break;
+        }
+        List<String> schemas = allSchemas
+                .stream()
+                .sorted(Comparator.comparingInt(Schema::getId))
+                .map(Schema::getSchemaDefinition)
+                .collect(Collectors.toList());
+        if (onlyLatest) {
+            // only latest
+            schemas = schemas.subList(schemas.size() - 1, schemas.size());
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("New schema {}", schemaDefinition);
+            for (String s : schemas) {
+                log.debug("Existing schema {}", s);
+            }
+        }
+        try {
+            CompatibilityExecutionResult compatibilityExecutionResult =
+                    checker.testCompatibility(level, schemas, schemaDefinition);
+            if (!compatibilityExecutionResult.isCompatible()) {
+                log.info("CompatibilityExecutionResult {}", compatibilityExecutionResult.isCompatible());
+                for (CompatibilityDifference error : compatibilityExecutionResult.getIncompatibleDifferences()) {
+                    log.info("CompatibilityExecutionResult error {}", error);
+                }
+            }
+            return compatibilityExecutionResult.isCompatible();
+        } catch (java.lang.IllegalStateException notSupported) {
```

Comment:
```suggestion
        } catch (IllegalStateException notSupported) {
```

[Codacy] `java.lang` package is imported implicitly.